src/lib/scheduler/src/thread_per_core.rs

```diff
@@ -129,6 +129,7 @@ where
     runner: TaskRunner<'pool, 'scope>,
 }
 
+#[allow(clippy::needless_lifetimes)]
```

Comment:
A comment about why we allow them here would be helpful. Same for the other override locations too.

Reply:
Added comments.